### ISSUE TITLE
Potential type error at `normalizeOptions(...)`

### ISSUE THREAD

**Issue by yfwz100** [jvilk/BrowserFS#227]
See: 

https://github.com/jvilk/BrowserFS/blob/bef1dc0b06021b04f66d90f2795e5cb8d25b7332/src/core/FS.ts#L79-L87

If `options` in `normalizeOptions(...)` is null, `typeof options` equals "object" and the following code will throw a `TypeError`:

https://github.com/jvilk/BrowserFS/blob/bef1dc0b06021b04f66d90f2795e5cb8d25b7332/src/core/FS.ts#L83

**Comment by yfwz100** [comment]
I found that #206 is a PR to this issue...

**Comment by yfwz100** [comment]
The upstream has updated the docs:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/81d0e4984d4b5e10668020eacd4b9a4542c5c837/types/node/index.d.ts#L4029

`fs.readFile()` accepts null as parameters for options.

**Comment by jvilk** [comment]
Yes; BrowserFS needs to be updated to track a specific version of the node FS API. It has gotten a bit out of sync. At the same time, I'm not happy with some of the code surrounding how BrowserFS emulates the Node API; it might be easier to simply repackage Node's JS modules and implement its internal bindings.

This is something I would like to revisit soon. However, I'm currently a bit busy preparing for a move across the country and finishing my PhD.

**Comment by yfwz100** [comment]
:+1:
